src/models/Query.py

```diff
@@ -15,10 +14,33 @@ def __init__(self, table: Table, columns: list[Column], condition: Condition):
             columns: columns to extract from the query
             condition: conditional statement to filter the results in the query
         """
+        self.table = table
+        self.columns = columns
+        self.condition = condition
 
     def to_SQL_string(self) -> str:
         """
         Transforms the query into SQL code.
         Returns:
             A string with the SQL code equivalent to the query.
         """
+        select_clause = ''
+        from_clause = ''
+        where_clause = ''
+        
+        if self.columns:
+            # SELECT statement
+            select_clause = f"SELECT {', '.join(column.name for column in self.columns)}"
+
+            if self.table:
+                # FROM statement
+                from_clause = f"FROM {self.table.name}"
+                
+                if self.condition:
+                    # WHERE statement (if condition exists)
+                    where_clause = f"WHERE {self.condition.to_SQL_string()}" if self.condition else ""
```

Comment:
Se recomienda no anidar if's cuando no sea necesario; en este caso no tendría ningún beneficio adicional no revisar los elementos de la tabla (líneas 35-37) en caso de que los elementos de las columnas no estén presentes (líneas 31-33). 

```suggestion
        if not self.table:
            # Raise appropiate exception

        # FROM statement
        from_clause = f"FROM {self.table.name}"
        
        # In case any of the next conditions is not true, keep the default behavior defined in lines 27-29
        if self.columns:
            # SELECT statement
            select_clause = f"SELECT {', '.join(column.name for column in self.columns)}"

        if self.condition:
            # WHERE statement (if condition exists)
            where_clause = f"WHERE {self.condition.to_SQL_string()}"
```